src/lib.rs

```diff
@@ -1472,6 +1530,85 @@ mod test {
         assert!(version().contains(&format!("{}.{}.{}", major, minor, patch)));
     }
 
+    #[test]
+    fn test_interrupt() {
+        use std::thread;
+        use std::time::Duration;
+        let tries = 15;
+        // Sadly, this is inherently finnicky. Even if sqlite gets the
+        // interrupt, it isn't guaranteed to stop. In practice, 15 tends to be
+        // more than enough so long as we move everything we can outside of
+        // the thread body, but it still is unfortunate.
+        for i in 0..tries {
+            let db = checked_memory_handle();
+            db.execute_batch("CREATE TABLE dummy(id)").unwrap();
+            let interrupt_handle = db.get_interrupt_handle();
+            // generate an arbitrary query which will be very slow to execute.
+            let sql = format!("{};",
+                (0..100_000).into_iter()
+                    .map(|i| format!("INSERT INTO dummy(id) VALUES({})", i))
+                    .collect::<Vec<_>>()
+                    .join(";\n"));
+
+            // Do this on the main thread to minimize the amount of time spent
+            // when interrupt won't do anything (because we haven't started
+            // executing the query).
+            let c_sql = str_to_cstring(&sql).unwrap();
+
+            let joiner = thread::spawn(move || {
+                unsafe {
```

Comment:
I am sorry but I don't understand why we should use unsafe code here ?

Reply:
It's to avoid the case where the sqlite_interrupt happens during the `str_to_cstring` done inside execute (it's a pretty big string...). In practice this happened a lot, and this made the test a decent amount more reliable.